ui/app/adapters/kmip/config.js

```diff
@@ -16,9 +16,19 @@ export default BaseAdapter.extend({
     return this._url(...arguments);
   },
   urlForCreateRecord(modelName, snapshot) {
-    return this._url(snapshot.id, modelName, snapshot);
+    const id = snapshot.id || snapshot.record.mutableId;
+    return this._url(id, modelName, snapshot);
   },
   urlForUpdateRecord() {
     return this._url(...arguments);
   },
+
+  createRecord(store, type, snapshot) {
+    return this._super(...arguments).then((response) => {
+      // if the server does not return an id and one has not been set on the model we need to set it manually from the mutableId value
+      if (!response?.id && !snapshot.record.id) {
+        return { id: snapshot.record.mutableId };
```

Comment:
could also set the id on snapshot.record and return snapshot.record here?

Reply:
Does the server ever return an id for this request? Presumably the `mutableId` will be set prior to calling `createRecord` so I'm wondering if checking for the id on the response or record is necessary and if the mutableId should always be returned?

Reply:
Good question! I just realize my comment didn't push, and some of this is copy pasta. The response here is null, so this should actually be. Buut...due to how irregular the api behaves, I think it'd make sense to first check if there is an id and use that if we can
```js
    return this._super(...arguments).then(() => {
      // the server doesn't not return anything on create, set an id to make ember-data happy
       return { id: snapshot.record.mutableId };
```